File: library/src/commonMain/kotlin/com/doordeck/multiplatform/sdk/api/responses/FusionResponses.kt
package com.doordeck.multiplatform.sdk.api.responses

import com.doordeck.multiplatform.sdk.api.model.UserRole
import kotlinx.serialization.Serializable
import kotlin.js.JsExport

@JsExport
@Serializable
class FusionLoginResponse(
    val authToken: String
)

@JsExport
@Serializable
class IntegrationTypeResponse(
    val status: String? = null
)

@JsExport
@Serializable
class DoorStateResponse(
    val state: ServiceStateType
)

@JsExport
@Serializable
class IntegrationConfigurationResponse(
    val doordeck: ControllerResponse? = null,
    val service: ServiceStateResponse? = null,
    val integration: DiscoveredDeviceResponse? = null
    // TODO MORE STUFF
)

@JsExport
@Serializable
class ControllerResponse(
    val id: String,
    val name: String? = null,
    val role: UserRole? = null
)

@JsExport
@Serializable
class ServiceStateResponse(
    val state: ServiceStateType
)

@JsExport
@Serializable
class DiscoveredDeviceResponse(
    //val key: Map<String, String>
)

@JsExport
@Serializable
enum class ServiceStateType {
    RUNNING,
    STOPPED,
    UNDEFINED
}

File: doordeck-sdk/src/mingwMain/resources/csharp/Model/Responses/FusionResponses.cs
using System.Text.Json.Serialization;

namespace Doordeck.Headless.Sdk.Model.Responses;

public class FusionLoginResponse
{
    public required string AuthToken { get; set; }
}
    
public class IntegrationTypeResponse
{
    public string? Status { get; set; }
}
    
public class DoorStateResponse
{
    public required ServiceStateType State { get; set; }
}
    
public class IntegrationConfigurationResponse
{
    public ControllerResponse? Doordeck { get; set; }
    public ServiceStateResponse? Service { get; set; }
    public DiscoveredDeviceResponse? Integration { get; set; }
}
    
public class ControllerResponse
{
    public required Guid Id { get; set; }
    public string? Name { get; set; }
    public UserRole? Role { get; set; }
}
    
public class ServiceStateResponse
{
    public required ServiceStateType State { get; set; }
}
    
public class DiscoveredDeviceResponse
{
    public required LockController Key { get; set; }
    public required Dictionary<string, string> Metadata { get; set; }
}

[JsonConverter(typeof(JsonStringEnumConverter))]
public enum ServiceStateType
{
    RUNNING,
    STOPPED,
    UNDEFINED
}